ui/components/ui/popover/popover.component.js

```diff
@@ -35,44 +78,31 @@ const Popover = ({
         {showArrow ? <div className="popover-arrow" /> : null}
         {showHeader && (
           <header className="popover-header">
-            <div
-              className={classnames(
-                'popover-header__title',
-                centerTitle ? 'center' : '',
-              )}
-            >
-              <h2 title="popover">
-                {onBack ? (
-                  <button
-                    className="fas fa-chevron-left popover-header__button"
-                    title={t('back')}
-                    onClick={onBack}
-                  />
-                ) : null}
-                {title}
-              </h2>
-              {onClose ? (
-                <button
-                  className="fas fa-times popover-header__button"
-                  title={t('close')}
-                  data-testid="popover-close"
-                  onClick={onClose}
-                />
-              ) : null}
-            </div>
-            {subtitle ? (
-              <p className="popover-header__subtitle">{subtitle}</p>
-            ) : null}
+            {hasStyles(headerProps) ? (
+              <Box {...headerProps}>
+                <Header />
+              </Box>
+            ) : (
+              <Header />
+            )}
           </header>
         )}
         {children ? (
           <div className={classnames('popover-content', contentClassName)}>
-            {children}
+            {hasStyles(contentProps) ? (
+              <Box {...contentProps}>{children}</Box>
+            ) : (
+              children
+            )}
           </div>
         ) : null}
         {footer ? (
           <footer className={classnames('popover-footer', footerClassName)}>
-            {footer}
+            {hasStyles(footerProps) ? (
+              <Box {...footerProps}>{footer}</Box>
+            ) : (
+              footer
+            )}
           </footer>
```

Comment:
Would we have specificity issues if we just changed the footer element to be a Box? e.g.

```
<Box {...footerProps} tag="footer" className={classnames('popover-footer', footerClassName, footerProps?.className)}>{footer}</Box>
```

Not sure if the optional chaining operator works or if it would list "undefined" as a className but you get the idea. I think we would need to add "footer" to `tag` prop options in `Box` too...

Reply:
Yeah, good point. I was on the fence of just straight up changing the tags themselves to `Box` but felt iffy on removing the tags, I had forgot about the `tag` prop. I think I'll add in the tags that are missing and change the elements to `Box` instead of nesting them.

Reply:
@georgewrmarshall actually, I think you might have been thinking about the `Typography` component, the `Box` component does not have a tag prop...I can add one though, but that opens the question of what tags we think should classify as a box. Also, why do you think the HTML rendering of `Box` only when there are actually props is confusing?

Reply:
Even then, I think it's fine to change the elements to `Box` themselves even though we don't specify a tag for them within the component itself..

Reply:
@hmalik88 yes you're right! I was. IMO all components should have a polymorphic tag prop that should allow any html element to be rendered as it's base. For type validation I think we would just add to the list as we go. Starting with `'div', 'footer'`. As to the second point. Personally, if I were to add a className or set of props using the `headerProps` or other prop object I would think that I'm adding it to the already existing root element without changing the DOM structure but that is just my opinion. Which ever pattern we choose I think we should just be consistent across the board.

Reply:
Oh I see your point! In that case, I'll take it out. Thanks!